lib/board.rb

```diff
@@ -85,14 +92,49 @@ def valid_placement?(ship, coordinates)
   end
 
 
+def place(ship, coordinates)
+
+  if valid_placement?(ship, coordinates)
+
+    coordinates.each do |coordinate|
+      @cells[coordinate].place_ship(ship)
+    end
+
+  else
+    "You can't put a ship here."
+  end
+end
+
+
+  def render(show_ships = false)
 
-  
+      first_row = "  1 2 3 4 \nA"
+      third_row = "\nB"
+      fourth_row = "\nC"
+      fifth_row = "\nD"
 
+      without_cells = [first_row, third_row, fourth_row, fifth_row]
+      with_cells = []
 
+      cells =  @cells.values
 
+      if show_ships == false
+        cells.each do |cell|
+        with_cells << cell.render
+        end
+     else
+        cells.each do |cell|
+        with_cells << cell.render(true)
+        end
+      end
+
+     rendered_cells = with_cells.each_slice(4).to_a
 
+     without_cells.zip(rendered_cells).flatten.join(' ')
```

Comment:
we eventually want a better way to accomplish this. as you mentioned before, we should start thinking of scalability.